### PR TITLE
Add interactive mode for update_dependencies

### PR DESCRIPTION
**Pull request by davidmreed**
# Critical Changes

# Changes

 - The `update_dependencies` and `install_managed` tasks accept the options `interactive` and `base-package-url-format`, which pause the automation and wait for the user to confirm the list of packages to be installed.

# Issues Closed

### CODE REVIEW

File: cumulusci/tasks/salesforce/install_package_version.py [hunk 1]
```diff
@@ -145,30 +165,45 @@ def _run_task(self):
...
+
+        if self.options.get("interactive", False):
+            if dep.version_id:
+                package_desc = self.options["base_package_url_format"].format(
+                    dep.version_id
+                )
+            else:
+                package_desc = str(dep)
+            self.logger.info("Package to install: {}".format(package_desc))
+            if not click.confirm("Continue to install dependencies?", default=True):
```
Comment (davisagli) [inline cumulusci/tasks/salesforce/install_package_version.py:184]: Should we try to validate the `interactive` option to throw an error if it's True but sys.stdin.isatty() is False?

Comment (davidmreed) [reply cumulusci/tasks/salesforce/install_package_version.py:184]: Seems reasonable, but I don't know if there are any Windows platform-specific issues with detecting an interactive environment. Any idea?

Comment (davidmreed) [reply cumulusci/tasks/salesforce/install_package_version.py:184]: Some quick googling suggests it false-positives on Windows.

Comment (davisagli) [reply cumulusci/tasks/salesforce/install_package_version.py:184]: Oh yeah, glad you thought to check that

**Review by jofsky — APPROVED** [review]
Looks good!

**Pull request merged**